src/LuaTransformer.ts

```diff
@@ -3085,7 +3358,7 @@ export class LuaTransformer {
         const leftType = this.checker.getTypeAtLocation(expression.left);
         this.validateFunctionAssignment(expression.right, rightType, leftType);
 
-        if (tsHelper.isArrayLengthAssignment(expression, this.checker, this.program)) {
+        if (tsHelper.isArrayLength(expression.left, this.checker, this.program)) {
```

Comment:
Same as above - this shouldn't be needed here now, right?

Reply:
I think this one is optimal. This turns an array length assignment into a LuaLib call instead of creating an iife that returns that LuaLib call's result.